internal/git/git.go

```diff
@@ -63,6 +68,35 @@ func (r *Repo) Git(args ...string) (string, error) {
 	return strings.TrimSpace(string(out)), nil
 }
 
+type RunOpts struct {
+	Args []string
+}
+
+type Output struct {
+	ExitCode int
+	Stdout   []byte
+	Stderr   []byte
+}
+
+func (r *Repo) Run(opts *RunOpts) (*Output, error) {
```

Comment:
New Git API because I want it to be more flexible/extensible. In the future will probably try to have the other one be something like `.RunSimple` which just returns a string and error.

Reply:
So this is same as above `func Git` but have Output object? Technically we can also just use this one everywhere?

Reply:
Yeah. I didn't wanna deal with refactoring everything right now just because the diff is already massive. But this is strictly more powerful than `.Git`.